### PR TITLE
Point send_test_email at the branch's custom SMTP provider

### PR DESCRIPTION
**Pull request by andrelandgraf**
## Problem

`configure_neon_auth send_test_email` required the SMTP password that the same tool refuses to give you.

`get_neon_auth_config` redacts it. A saved standard provider comes back as `password: "***redacted***"`, and the Neon API itself returns an empty string for that field on GET for ordinary callers. So the sequence an agent naturally runs, save a provider with `update_email_provider` then check that it delivers, had no password to pass into the test call. The sentinel and the empty string both fail authentication at the relay, so the only working path was to ask the person to paste the password a second time.

The snapshot had a related defect. `emailProviderSnapshot` mapped `password: data.password ? REDACTED_SECRET : null`, and the file's contract is that the sentinel always means the secret is set and `null` always means it is not. Because the API returns `""`, every configured standard provider was reported as having no password:

```json
"email_provider": {
  "type": "standard",
  "host": "smtp.sendgrid.net",
  "port": 587,
  "username": "apikey",
  "password": null,
  "sender_email": "noreply@example.com",
  "sender_name": "Acme"
}
```

## Diagnosis

Empty is the redaction, not an absence. `@neon/sdk` 2.2.0 documents the field on `StandardEmailServerResponse`:

> On GET, returned redacted (empty) for ordinary callers, while callers with project-credential read permission receive the stored password. Update (PATCH) responses always return it redacted (empty) regardless of permission.

Both halves of the fix were waiting on that SDK bump rather than on a design decision here. The same release ships `sendNeonAuthEmailProviderTest`, a POST to `/projects/{project_id}/branches/{branch_id}/auth/email_provider/test` that takes `recipient_email` and reads the stored SMTP settings server-side. Its generated doc names the exact failure above as the reason it exists: it avoids the masked password being sent back, which would fail SMTP authentication. The old `sendNeonAuthTestEmail` is deprecated in 2.2.0 and stays available for testing an unsaved configuration and for non-Better-Auth providers.

## The interface

`test_email` now takes one field. The schema is `.strict()`, so the old argument shape is a validation failure rather than a silently ignored extra.

```json
{
  "operation": "send_test_email",
  "projectId": "proj-1",
  "test_email": { "recipient_email": "tester@example.com" }
}
```

`host`, `port`, `username`, `password`, `sender_email` and `sender_name` are no longer accepted here. They remain required on `update_email_provider` for a `type: "standard"` provider, which is unchanged.

The request on the wire carries the recipient and nothing else:

```http
POST /projects/proj-1/branches/br-1/auth/email_provider/test
Authorization: Bearer <api key>
Content-Type: application/json

{"recipient_email":"tester@example.com"}
```

Success text:

```
Test email dispatched to tester@example.com using the custom SMTP provider saved on this branch.
```

A 200 that reports `success: false` sets `isError` and passes the upstream reason through verbatim:

```
Test email could NOT be sent to tester@example.com using the custom SMTP provider saved on this branch.
Upstream error: <error_message from the API>
```

A rejected dispatch arrives as a `NeonApiError` and renders through `handleToolError` in the existing envelope:

```
[HTTP <status>] <upstream message>
```

The message inside that envelope is the API's own text. It has not been captured from a live branch, so the exact wording for a missing provider, a shared provider or a non-Better-Auth integration is not documented here.

`get_neon_auth_config` reports a standard provider's password as the sentinel unconditionally. The mapping no longer reads `data.password`, so what the API returned for that field cannot reach the caller either way:

```json
"email_provider": {
  "type": "standard",
  "host": "smtp.sendgrid.net",
  "port": 587,
  "username": "apikey",
  "password": "***redacted***",
  "sender_email": "noreply@example.com",
  "sender_name": "Acme"
}
```

Nothing else in the snapshot changes. The `shared` branch, OAuth `client_secret` redaction and every other key are as they were, and `send_test_email` still does not reload the snapshot, since dispatching a message mutates no Neon Auth state.

## Also in here

- `@neon/sdk` goes from 2.0.0 to 2.2.0, with `pnpm-lock.yaml`. That release is what supplies `sendNeonAuthEmailProviderTest` and the `NeonAuthEmailServerConfigResponse` type the snapshot now reads.
- The snapshot's email-provider fetch and builder switch from `NeonAuthEmailServerConfig` to `NeonAuthEmailServerConfigResponse`, which is the type that actually describes a GET body.
- Three unit assertions that checked the supplied SMTP password never appeared in the rendered output are gone. They tested a value the schema no longer accepts, so there is no password in scope to leak. The wire-level test asserts the full request body instead, which covers the same ground from the other side.
- `standardEmailServerFields` is now used only by `update_email_provider`. The comment saying both operations reuse it was removed because it was no longer true.
- `CHANGELOG.md` gets a `[NEXT]` entry stating the new argument shape, the schema rejection and the snapshot redaction, since a caller of the published server hits all three.

## Verification

Against base `ea9259a`:

- `pnpm test:unit`: 513 tests across 29 files, all pass.
- `pnpm test:integration`: 120 pass. 9 remain skipped in `session-binding.integration.test.ts`, which this branch does not touch.
- `pnpm typecheck` and `pnpm lint`: clean.

Behaviours covered by the tests in this diff:

- `test_email` with `recipient_email` alone parses
- `test_email` carrying the old SMTP fields fails schema validation
- a malformed `recipient_email` fails schema validation
- the handler posts exactly `{"recipient_email": "..."}` to the saved-provider path with the bearer token, asserted against a real HTTP server on loopback rather than a stub
- a 200 with `success: false` returns `isError` and surfaces the upstream `error_message`
- a resolved non-200 returns `isError` and surfaces `error_message` when the body carries one
- a 400 from the endpoint reaches the caller as a `NeonApiError` and renders through `handleToolError` as the `[HTTP <status>]` envelope
- a 502 reaches the caller as a `NeonApiError` with the status preserved
- `send_test_email` does not touch the snapshot fetchers
- a GET standard provider with `password: ""` snapshots as `***redacted***`

Not verified: no live Neon branch with a saved custom SMTP provider was exercised. Delivery through a real relay and the API's actual rejection text for each unsupported configuration are both unconfirmed. The integration test drives a local HTTP server standing in for the Neon API, so it proves the request shape and the error plumbing, not the upstream behaviour.

## For your attention

- **This breaks any caller passing the old arguments.** `.strict()` turns the previous request into a validation error rather than dropping the extra fields. The changelog entry says so.
- **Testing an unsaved SMTP configuration is no longer reachable through this tool.** The provider has to be saved with `update_email_provider` first. The deprecated endpoint that accepts full credentials still exists upstream and is no longer called from here.
- **The standard-provider snapshot no longer distinguishes a set password from an unset one.** `update_email_provider` requires `password` for `type: "standard"`, so a saved standard provider without one is not a state the API can be in. The `EmailProviderSnapshot` type still allows `null` on that field for the shape's sake, and the standard branch never produces it.
- **The three rejection cases are enforced by the API, not by the schema.** A shared provider, a missing configuration and a non-Better-Auth integration all pass validation here and come back as a client error. The tool description and the schema description name them so the model does not have to discover them by failing.

### CODE REVIEW

File: mcp/tools/handlers/neon-auth-settings-snapshot.ts [hunk 1]
```diff
@@ -140,7 +135,7 @@ function oauthProviderSnapshot(
 }
 
 function emailProviderSnapshot(
-  data: NeonAuthEmailServerConfig,
+  data: NeonAuthEmailServerConfigResponse,
```
Comment (Shridhad) [inline mcp/tools/handlers/neon-auth-settings-snapshot.ts:138]: **Important:** switching this to `NeonAuthEmailServerConfigResponse` is correct for GET, but the mapper a few lines down is still `password: data.password ? REDACTED_SECRET : null`.

Upstream `StandardEmailServerResponse.password` is required and **empty when redacted** for ordinary callers (privileged keys may still get the real value). Empty string is falsy, so a saved standard provider is snapshotted as `password: null` instead of `***redacted***`. Agents will treat SMTP as unset and may try to “fix” it.

For `type === 'standard'`, treat empty as redacted (a standard provider is not valid without a stored password). Please add a unit case with `password: ''` — current tests only feed a non-empty sentinel.

**Review by Shridhad — CHANGES_REQUESTED** [review]
## Review

`send_test_email` itself looks right: recipient-only body, strict schema so leftover SMTP fields fail validation, `throwOnError` / `success: false` → `isError`, and the wire test pins the new endpoint. Dropping credentials from the test path is a real improvement.

**Requesting changes** for the GET snapshot mapping that this SDK type change made incorrect for ordinary callers.

### Important
- `emailProviderSnapshot` still treats a falsy `password` as unset. `NeonAuthEmailServerConfigResponse` / `StandardEmailServerResponse` returns an **empty string** when the stored password is redacted. That snapshots a configured standard provider as `password: null`, which contradicts the `***redacted***` = set / `null` = unset contract and can send agents into a re-PATCH loop. Tests only mock a non-empty sentinel.

### Consider
- Result copy still says “SMTP provider saved on this branch” while the tool description says custom SMTP (`type=standard`). Shared-provider 4xx uses the same throw path as missing config, so a distinct fixture is optional.
- Fat-tool `openWorldHint: false` is pre-existing; not blocking.

Happy to approve once empty GET passwords map to `***redacted***` and there is a `password: ''` unit case.

File: mcp/tools/handlers/neon-auth-settings-snapshot.ts (hunk 1 above)
Comment (andrelandgraf) [reply mcp/tools/handlers/neon-auth-settings-snapshot.ts:138]: Andre's agent: For `type === 'standard'`, empty GET `password` now maps to `***redacted***` (standard always has a stored password). Added a `password: ''` unit case on `get_neon_auth_config`. Result copy now says "custom SMTP provider saved on this branch", matching the tool description. Same result-copy wording is on the CLI PR.

**Review by Shridhad — APPROVED** [review]
## Re-review

Follow-up `02e90e8` addresses the request-changes item.

- Standard GET snapshots now always use `***redacted***` (`password: REDACTED_SECRET`), so `""` is no longer reported as unset, and a privileged caller’s real password is still never echoed.
- `get_neon_auth_config` has a `password: ''` case that asserts the sentinel.
- Result copy now says “custom SMTP provider saved on this branch”, matching the tool description.

`send_test_email` wiring, strict schema, and the wire test still look good. `lint-and-build` is green. Approving.
